_posts/2020-07-21-cube-functionality-to-kartothek.markdown

```diff
@@ -0,0 +1,409 @@
+---
+layout: single
+title:  "Introducing Cube Functionality To Kartothek"
+date:   2020-07-21 06:00:00 +0500
+tags: technology python data-engineering
+header:
+  overlay_image: assets/images/tech_gear_banner.jpg
+  overlay_filter: 0.2
+  show_overlay_excerpt: false
+author: Usha Nemani & Marco Neumann
+author_profile: true
+---
+# Introducing Cube Functionality To Kartothek
+
+Last year, we introduced [Kartothek](../introducing-kartothek/), a table management python library powered by Dask. 
+Our journey continued by adding another gem to our open source treasure: We empowered Kartothek with multi-dataset functionality.
+ 
+You might think: "Kartothek already provides dataset features, do I really need multiple dataset interfaces?" Hang on,
+ we will present you the story of **Kartothek cubes**, an interface that supports multiple datasets. 
+
+Imagine a typical machine learning workflow, which might look like this:
+* First, we get some input data, or source data. In the context of Kartothek cubes, we will refer to the source data as seed data or seed dataset.
+* On this seed dataset, we might want to train a model that generates predictions.
+* Based on these predicitons, we might want to generate reports and calculate KPIs.
+* Last, but not least, we might want to create some dashboards showing plots of the aggregated KPIs as well as the underlying input data.
+What we need for this workflow is not a table-like view on our data, but a view on everything that we generated in these different steps.
+ 
+Kartothek Cubes deals with multiple Kartothek datasets, loosely modeled after [Data Cubes](https://en.wikipedia.org/wiki/Data_cube).
+
+Cubes offer an interface to query all of the data without performing complex join operations manually each time. 
+
+Because kartothek offers a view on our cube as a large virtual pandas DataFrame, querying the whole dataset is very comfortable.
```

Comment:
```suggestion
Because kartothek offers a view on our cube similar to large virtual pandas DataFrame, querying the whole dataset is very comfortable.
```

"as a" somewhat implies that the functionality / interface is modeled after the pandas DataFrame, which is not true (Dask.DataFrame for example does this though)